src/utils.js

```diff
@@ -24,18 +26,18 @@ var t_Arr = 'Array',
  *   map['something'] = 'something else';
  *   console.log(replaceTokenInString(str, map, '%%')); => "text it was subbed this text with something else"
  */
+var tokenRxpMap = {},
+    wordMatch = '(\\w+)';
+
+function getRegexpForToken(token) {
+  return (tokenRxpMap[token] ? tokenRxpMap[token] : (tokenRxpMap[token] = new RegExp(token + wordMatch + token, 'g')));
+}
+
 exports.replaceTokenInString = function(str, map, token) {
-	for (var key in map) {
-		if (map.hasOwnProperty(key)) {
-			var keyString = token + key.toUpperCase() + token;
-			if (typeof map[key] === objectType_undefined) {
-				map[key] = '';
-			}
-			var re = new RegExp(keyString, 'g');
-			str = str.replace(re, map[key]);
-		}
-	}
-	return str;
+  var regex = getRegexpForToken(token);
```

Comment:
There seems to be some issues with this function. It's not working the way it was previously. Can you revert it for now and make some test cases.